cosmo_tester/test_suites/bootstrap_based_tests/distributed_installation_test.py

```diff
@@ -112,38 +185,179 @@ def test_distributed_installation_scenario(database_and_manager,
+    # Pick a manager to operate on, and trust the cluster to work with us
+    manager = distributed_installation.instances[0]
+
+    hws = []
+    for i in range(0, 2):
+        tenant = prepare_and_get_test_tenant(
+            'clusterhello{num}'.format(num=i),
+            manager,
+            cfy,
+        )
+        hw = centos_hello_world(
+            cfy, manager, attributes, ssh_key, logger, tmpdir,
+            tenant=tenant, suffix=str(i),
+        )
+        hws.append(hw)
+
+    yield hws
+    for hw in hws:
+        if hw.cleanup_required:
+            logger.info('Cleaning up hello world...')
+            manager.use()
+            hw.cleanup()
+
+
+@pytest.fixture(scope='function')
+def distributed_nodecellar(cfy, distributed_installation, attributes,
+                           ssh_key, tmpdir, logger):
+    # Uploading to the manager not the database
+    manager = distributed_installation.manager
     manager.use()
-    logger.info('Using manager `{0}`'.format(manager.ip_address))
-    cfy.profiles.set('-u', USER_NAME, '-p', USER_PASS, '-t', TENANT_NAME)
+    tenant = prepare_and_get_test_tenant(TENANT_NAME, manager, cfy)
+    nc = NodeCellarExample(
+        cfy, manager, attributes, ssh_key, logger, tmpdir,
+        tenant=tenant, suffix='simple')
+    nc.blueprint_file = 'simple-blueprint-with-secrets.yaml'
+    yield nc
 
 
-def _set_admin_user(cfy, manager, logger):
+def _set_test_user(cfy, manager, logger):
     manager.use()
     logger.info('Using manager `{0}`'.format(manager.ip_address))
-    cfy.profiles.set('-u', 'admin', '-p', 'admin', '-t', 'default_tenant')
+    cfy.profiles.set('-u', USER_NAME, '-p', USER_PASS, '-t', TENANT_NAME)
 
 
-def _create_secrets(cfy, logger, attributes, manager):
-    logger.info('Creating secret agent_user as blueprint input')
-    cfy.secrets.create('agent_user', '-s', attributes.default_linux_username,
-                       visibility='global')
+def disable_enable_node(manager, logger, disable=True):
+    """
+    Disable or enable a manager to avoid it from being picked as the leader
+    during tests
+    """
+    action_msg, action = \
+        ("Shutting down", 'stop') if disable else ("Starting", 'start')
+    with manager.ssh() as fabric:
+        logger.info('{0} nginx service on manager {1}'.format(
+            action_msg, manager.ip_address))
+        fabric.run('sudo systemctl {0} nginx'.format(action))
 
-    logger.info('Creating secret agent_private_key_path as blueprint input')
-    cfy.secrets.create('agent_private_key_path', '-s',
-                       manager.remote_private_key_path, visibility='global')
 
-    logger.info('Creating secret host_ip as blueprint input')
-    cfy.secrets.create('host_ip', '-s', manager.ip_address,
-                       visibility='global')
+def reverse_cluster_test(cfy, cluster_machines, logger):
+    for manager in cluster_machines.instances:
+        disable_enable_node(manager, logger, disable=False)
```

Comment:
i'd reverse the naming, making it `enable_disable_node(..., enable=True)` because the double negative in "disable=False" makes it harder to follow

also i guess maybe there can be a better name than "enable_disable" (maybe "toggle" or something)

Reply:
Fixed